src/module-elasticsuite-catalog/Model/Product/Search/Request/Container/Filter/Visibility.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * DISCLAIMER
+ * Do not edit or add to this file if you wish to upgrade Smile Elastic Suite to newer
+ * versions in the future.
+ *
+ * @category  Smile
+ * @package   Smile\ElasticsuiteCatalog
+ * @author    Romain Ruaud <romain.ruaud@smile.fr>
+ * @copyright 2018 Smile
+ * @license   Open Software License ("OSL") v. 3.0
+ */
+namespace Smile\ElasticsuiteCatalog\Model\Product\Search\Request\Container\Filter;
+
+use Smile\ElasticsuiteCore\Api\Search\Request\Container\FilterInterface;
+use Smile\ElasticsuiteCore\Search\Request\QueryInterface;
+
+/**
+ * Product Visibility Default filter.
+ *
+ * @category Smile
+ * @package  Smile\ElasticsuiteCatalog
+ * @author   Romain Ruaud <romain.ruaud@smile.fr>
+ */
+class Visibility implements FilterInterface
```

Comment:
Bad logic.
You should not have an if but 2 classes instead.

The visibility is not defined by having or not a search query but by the container itself.